Lib/test/test_buffer.py

```diff
@@ -37,7 +37,8 @@
     ctypes = None
 
 try:
-    with warnings.catch_warnings():
+    with support.EnvironmentVarGuard() as os.environ, \
```

Comment:
Does it restore os.environ to its original value after the with block? Do we need the "as" part?

Reply:
Yes, it restores os.environ to its original value after the with block. EnvironmentVarGuard() just creates a proxy to os.environ, it doesn't touch os.environ, and it doesn't track direct modifications of os.environ. We need the "as" part for making all changes done through the proxy.

Reply:
Oh, EnvironmentVarGuard doesn't set itself as os.environ, but its .__exit__() method restores os.environ to the previous object. I see.